extensions/amp-base-carousel/1.0/base-carousel.type.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @externs */
+
+/** @const */
+var BaseCarouselDef = {};
+
+/**
+ * @typedef {{
+ *   arrowNext: (?PreactDef.VNode|undefined),
+ *   arrowPrev: (?PreactDef.VNode|undefined),
+ *   children: (!PreactDef.Renderable),
+ *   defaultSlide: (number|undefined),
+ *   loop: (boolean|undefined),
+ *   slide: (number|undefined),
+ *   onMount: (function():undefined|undefined),
+ *   onSlideChange: (function(number):undefined|undefined),
+ *   setAdvance: (function(function(number):undefined):undefined|undefined),
+ * }}
+ */
+BaseCarouselDef.Props;
+
+/**
+ * @typedef {{
+ *   children: !Array<PreactDef.Renderable>,
+ *   ignoreProgrammaticScroll: {current: (boolean|null)},
```

Comment:
Is this a ref? I'd prefer we suffix refs with `Ref` since it's a pretty internal thing per a component most of the time.